### DISCUSSION TITLE
Run behind reverse proxy

### DISCUSSION THREAD

**Discussion by felixhummel** [harlan-zw/unlighthouse#95] [Q&A] [answered]
I tried running unlighthouse behind a reverse proxy at https://unlighthouse.example.com/ with the following Caddy config:
```
https://unlighthouse.example.com/ {  
  reverse_proxy http://localhost:5678  
}  
```

Please note that [Caddy automatically sets `X-Forwarded-` headers](https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#defaults) like `X-Forwarded-Host` and `X-Forwarded-Proto`.
                                                                             
Unfortunately the UI tries to access the API at http://localhost:5678/api/scan-meta nonetheless (as seen in the DevTools' network tab).                                     

While I believe that unlighthouse server should respect `X-Forwarded-` headers,
it would suffice if one could set the *server*'s base URL.

Is there some way to configure this?
Should I open an issue?

**Comment by harlan-zw** [comment]
Hey @felixhummel, thanks for the question :)

Sorry, but I don't have an immediate answer for this and have limited capacity to investigate further. Any help would be appreciated otherwise I'll try to get back to you when I can dig into it.

**Comment by jurgenhaas** [comment]
I'm having the same issue, although I'm using a different proxy: Traefik. But the problem is the same.

Just wondering if, instead of respecting `X-Forwarded-`headers, an alternative solution could be an optional config argument that would allow us to define the base URL manually. This is fairly important, since otherwise we get blocked in modern browsers because of cross-site scripting prevention.

**Comment by jurgenhaas** [comment, marked as answer]
Ah, it looks like this works by adding these options to the config file:

```
    serverUrl: "https:/unlighthouse.example.com/",
    clientUrl: "https://unlighthouse.example.com/",
    apiUrl: "https://unlighthouse.example.com/api",
    websocketUrl: "wss:///unlighthouse.example.com/api/ws",
```

**Reply by felixhummel** [reply]
Thanks! Slashes in your example are a little off. Here's what works for me:

```
export default {
  site: "https://example.com",
  url: "https://unlighthouse.example.com",
  apiUrl: "https://unlighthouse.example.com/api",
  websocketUrl: "wss://unlighthouse.example.com/api/ws",
  serverUrl: "https://unlighthouse.example.com/",
  clientUrl: "https://unlighthouse.example.com/",
}
```
